placement/algo/mirrored.go

```diff
@@ -205,12 +242,54 @@ func (a mirroredAlgorithm) ReplaceInstances(
 	return p, nil
 }
 
+// allInitializing returns true when the given list of instances
+// matches all the initializing instances in the placement.
+func allInitializing(instances []string, p placement.Placement) bool {
+	ids := make(map[string]struct{}, len(instances))
+	for _, i := range instances {
+		ids[i] = struct{}{}
+	}
+
+	for _, i := range p.Instances() {
+		if !i.IsInitializing() {
+			continue
+		}
+		if _, ok := ids[i.ID()]; !ok {
+			return false
+		}
+		delete(ids, i.ID())
```

Comment:
nit: Take it or leave it, but use a `var totalInstances = len(instances)` and decrement that count every time there's a match is probably cheaper than deleting an entry from the map.